src/client/java/tech/thatgravyboat/skyblockapi/mixins/tagattachments/ClientPacketListenerMixin.java

```diff
@@ -0,0 +1,44 @@
+package tech.thatgravyboat.skyblockapi.mixins.tagattachments;
+
+import com.llamalad7.mixinextras.sugar.Local;
+import net.minecraft.client.multiplayer.ClientPacketListener;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.protocol.game.ClientboundSetEntityDataPacket;
+import net.minecraft.network.syncher.SynchedEntityData;
+import net.minecraft.world.entity.Entity;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import tech.thatgravyboat.skyblockapi.api.SkyBlockAPI;
+import tech.thatgravyboat.skyblockapi.api.events.entity.ListenForNameChange;
+import tech.thatgravyboat.skyblockapi.api.events.entity.NameChangedEvent;
+import tech.thatgravyboat.skyblockapi.mixins.accessors.EntityAccessor;
+
+import java.util.Optional;
+
+@Mixin(ClientPacketListener.class)
+public class ClientPacketListenerMixin {
```

Comment:
Dont do this in here, do it inside of Entity#onSyncedDataUpdated as this can happen in more than 1 place